Tools/autotest/vehicle_test_suite.py

```diff
@@ -2559,64 +2559,51 @@ def test_adsb_send_threatening_adsb_message(self, here, offset_ne=None):
     def get_sim_parameter_documentation_get_whitelist(self):
         # common parameters
         ret = set([
-            "SIM_ACC_FILE_RW",
-            "SIM_ACC_TRIM_X",
-            "SIM_ACC_TRIM_Y",
-            "SIM_ACC_TRIM_Z",
-            "SIM_ADSB_ALT",
-            "SIM_ADSB_COUNT",
-            "SIM_ADSB_RADIUS",
-            "SIM_ADSB_TX",
-            "SIM_ARSPD2_OFS",
-            "SIM_ARSPD2_RND",
-            "SIM_ARSPD_OFS",
-            "SIM_ARSPD_RND",
-            "SIM_BAR2_DELAY",
-            "SIM_BAR2_DISABLE",
-            "SIM_BAR2_DRIFT",
-            "SIM_BAR2_FREEZE",
-            "SIM_BAR2_WCF_BAK",
-            "SIM_BAR2_WCF_DN",
-            "SIM_BAR2_WCF_FWD",
-            "SIM_BAR2_WCF_LFT",
-            "SIM_BAR2_WCF_RGT",
-            "SIM_BAR2_WCF_UP",
-            "SIM_BAR3_DELAY",
-            "SIM_BAR3_DISABLE",
-            "SIM_BAR3_DRIFT",
-            "SIM_BAR3_FREEZE",
-            "SIM_BAR3_WCF_BAK",
-            "SIM_BAR3_WCF_DN",
-            "SIM_BAR3_WCF_FWD",
-            "SIM_BAR3_WCF_LFT",
-            "SIM_BAR3_WCF_RGT",
-            "SIM_BAR3_WCF_UP",
-            "SIM_BARO_COUNT",
-            "SIM_BARO_DELAY",
-            "SIM_BARO_DISABLE",
-            "SIM_BARO_FREEZE",
-            "SIM_BARO_WCF_BAK",
-            "SIM_BARO_WCF_DN",
-            "SIM_BARO_WCF_FWD",
-            "SIM_BARO_WCF_LFT",
-            "SIM_BARO_WCF_RGT",
-            "SIM_BARO_WCF_UP",
-            "SIM_BATT_CAP_AH",
-            "SIM_BAUDLIMIT_EN",
-            "SIM_DRIFT_SPEED",
-            "SIM_DRIFT_TIME",
-            "SIM_EFI_TYPE",
-            "SIM_ESC_ARM_RPM",
-            "SIM_FTOWESC_ENA",
-            "SIM_FTOWESC_POW",
+            "SIM_ACC_TRIM_X",    # not listed in SITL.cpp
+            "SIM_ACC_TRIM_Y",    # not listed in SITL.cpp
+            "SIM_ACC_TRIM_Z",    # not listed in SITL.cpp
+            "SIM_ARSPD2_OFS",    # not listed in SITL.cpp
+            "SIM_ARSPD2_RND",    # not referenced anywhere
+            "SIM_ARSPD_OFS",     # not referenced anywhere
+            "SIM_ARSPD_RND",     # not listed in SITL.cpp
+            "SIM_BAR2_DELAY",    # not referenced anywhere
+            "SIM_BAR2_DISABLE",  # not listed in SITL.cpp
+            "SIM_BAR2_DRIFT",    # not referenced anywhere
+            "SIM_BAR2_FREEZE",   # not listed in SITL.cpp
+            "SIM_BAR2_WCF_BAK",  # not referenced anywhere
```

Comment:
Everything in this list is used - we have a sanity check for that ;-)

In this case the parameters are coming from in here: https://github.com/ardupilot/ardupilot/blob/pr%2Fstop-using-python-is-python3/libraries/SITL/SITL.cpp#L589

Fixing this requires moving the BaroParm structure to a new file, probably SITL_Baro.cpp .  @antholuo did the same thing to document airspeed by creating SITL_Airspeed.cpp, so use that as a model

Reply:
Thanks I can follow the approach in #25005. Do you want that to be a separate PR or included in this one?

Reply:
Separate PR makes sense.  Makes these easy to review!